package/network/services/cifsd-tools/files/cifsd.init

```diff
@@ -0,0 +1,56 @@
+#!/bin/sh /etc/rc.common
+
+START=98
+
+# FIXME: We get a router crash/restart using procd via 'cifsd -n', because of the two worker, manager processes ?
+
+start(){
+
+	if [ ! -f /etc/cifs/smb.conf ]; then
+		logger -t 'cifsd' "missing config /etc/cifs/smb.conf, needs to-be created manually!"
+		exit 1
+	fi
+
+	[ -f /tmp/cifsd.lock ] && rm /tmp/cifsd.lock
+	
+	# try remove again before start
+	if (lsmod | grep cifsd &>/dev/null); then
+		rmmod cifsd &>/dev/null
+	fi
+	modprobe cifsd 2>/dev/null
```

Comment:
Why is it needed to remove the kernel module before this gets started?

Reply:
Some setting may not be fully reflected without a full remove, so i just added a extra try before start.